src/ucp/tag/offload.h

```diff
@@ -85,4 +88,53 @@ ucp_tag_offload_try_cancel(ucp_worker_t *worker, ucp_request_t *req, int force)
     }
 }
 
+/* For homogeneous clusters it is expected that tag offload messages will arrive
+ * from the single interface. Otherwise need to add a particular iface to tags
+ * hash, where key is a tag masked with tag_sender mask. */
+static UCS_F_ALWAYS_INLINE void
+ucp_tag_offload_unexp(ucp_worker_iface_t *iface, ucp_tag_t tag)
```

Comment:
looks like we have 3 extra checks for every unexp eager message, need to avoid it in case the interface is already active